magento/magento2
# 12696 Delete all test modules after integration tests

### Description
Several test modules are created on the fly during startup of every
integration test run, but they are never deleted. This PR
changes this, modules are deleted when the process ends.

Things we thought about but didn't implement:

- Putting the modules into `dev/tests/integration/tmp/*/app/code/`: we'd need to modify the autoloader and the component registrar which might have caused side effects. Plus, the approach we tried didn't work out.
- Leave the modules in `dev/tests/integration/_files` and register them from there: same as above.
- Remove the modules from a `dev/tests/integration/framework/removeTestModules.php`: PhpUnit doesn't offer an option to run a script after all tests have been executed.

### Fixed Issues
1. magento/magento2#12696: Integration tests create stub modules in app/code

### Manual testing scenarios
1. Run any integration test
2. Verify that during the test run, some directories names `TestModule*`, i.e. `TestModuleSample`, exist in `app/code/Magento` (they are created automatically when the tests are started).
3. Verify that those directories are gone after the tests have ended.

### Contribution checklist
